test/blocks/global-navigation/utilities/utilities.test.js

```diff
@@ -26,27 +28,69 @@ describe('global navigation utilities', () => {
     expect(fragment2.tagName).to.equal('SPAN');
   });
 
+  // TODO - no tests for using the the live url and .hlx. urls
+  // as mocking window.location.origin is not possible
+  describe('getFedsContentRoot', () => {
+    it('should return content source for localhost', () => {
+      const contentSource = getFedsContentRoot();
+      expect(contentSource).to.equal('https://main--milo--adobecom.hlx.page');
+    });
+  });
+
+  describe('federatePictureSources', () => {
+    it('should use federated content root for image sources', async () => {
```

Comment:
You could again consider splitting the tests to make them a little more robust and separating the cases... but up to your preference, it's also good as is.
Demo code:
```
it('should update srcset for source elements', () => {
    const sourceElement = document.createElement('source');
    sourceElement.setAttribute('srcset', `./${imgPath}`);
    federatePictureSources(sourceElement);
    expect(sourceElement.getAttribute('srcset')).to.equal(`https://main--milo--adobecom.hlx.page/${imgPath}`);
});

it('should update src for img elements', () => {
    const imgElement = document.createElement('img');
    imgElement.setAttribute('src', `./${imgPath}`);
    federatePictureSources(imgElement);
    expect(imgElement.getAttribute('src')).to.equal(`https://main--milo--adobecom.hlx.page/${imgPath}`);
});
```

Reply:
Well usually the templates have both. The issue with images is around the ones added directly in the Word document, as they're relative by default. They're added as `picture` tags, with both `source` and `img` children. So I'd argue this is testing the real scenario, by combining both.